File: 05 - filas/java/Fila2.java
import java.util.LinkedList;
import java.util.Queue;

// Estruturas de Dados
//     - físicas: variáveis
//     - abstratas: pilha(stack), lista(list), fila(queue)
//          pilha: push/add/append e pop/remove -> topo
//          lista: add/append, remove, contains, sort -> ordenação e controle de duplicidade
//          fila: add/append (final) e remove (inicio)
// FILA - FIFO - First In, First Out

public class Fila2 {
    public static void main(String[] args) {
        Queue<String> filaNomes = new LinkedList<String>();

        filaNomes.add("Saccol");
        filaNomes.add("Pinheiro");
        filaNomes.add("Iago");

        System.out.println("Exibindo fila");
        System.out.println(filaNomes);

        System.out.println("Exibindo inicio ou cabeça/head da fila");
        System.out.println(filaNomes.peek());

        System.out.println("Exibindo se fila esta vazia");
        System.out.println(filaNomes.isEmpty());

        System.out.println("Exibindo o tamanho da fila");
        System.out.println(filaNomes.size());

        System.out.println("Verificando se IAGO esta na fila");
        System.out.println(filaNomes.contains("IAGO"));
 
        while (!filaNomes.isEmpty()){
            System.out.println("Removendo ..." + filaNomes.poll());
            try {
                Thread.sleep(3000);
            } catch(Exception e){
                System.out.println(e.getMessage());
            }
        }

        System.out.println("Removendo ..." + filaNomes.poll());
    }
}

File: 05 - filas/python/fila2.py
import queue 
import time

from aluno import Aluno

fila_alunos = queue.Queue()

fila_alunos.put(Aluno(1,"Saccol"))
fila_alunos.put(Aluno(7,"Pinheiro"))
fila_alunos.put(Aluno(2,"Iago"))

for i in range(2):    
    matricula = input("Matricula: ")
    nome = input("Nome: ")
    fila_alunos.put(Aluno(matricula, nome))

print("Exibindo fila")
for item in fila_alunos.queue:
    print(item)

print("Exibindo inicio ou cabeça/head da fila")
print(fila_alunos.queue[0])

print("Exibindo final da fila")
print(fila_alunos.queue[-1])

print("Exibindo se fila esta vazia")
print(fila_alunos.empty())

print("Exibindo o tamanho da fila")
print(fila_alunos.qsize())    

print("Verificando se 2,Iago esta na fila")
a = Aluno(2,"Iago")
print(a in fila_alunos.queue)

while (not fila_alunos.empty()):
    print("Removendo ...")
    print(str(fila_alunos.get_nowait()))
    time.sleep(1)
    
# print("Removendo ..." + str(fila_alunos.get_nowait()))
